From 42669668f9f6bbf1d8fa6d50b0faf8ca896509f1 Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Fri, 14 Nov 2014 09:00:24 +0800
Subject: [PATCH] preempt-rt/softirq: Fix the operand warning

Since the commit 0a29f66056cd [tasklet: fix a race condition in the
state-bit flipping in softirq.c] using '==' and '|' in the
__tasklet_common_schedule function. The origin mean is to equal
the variable | TASKLET_STATEF_RUN. So add a pair of parentheses
to fix the warning.

The warning information as the following:
kernel/softirq.c:
In function '__tasklet_common_schedule':
kernel/softirq.c:845:5:
warning: suggest parentheses around comparison in operand of '|'
[-Wparentheses]

Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 kernel/softirq.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/kernel/softirq.c b/kernel/softirq.c
index 70ded1f..cfe11d1 100644
--- a/kernel/softirq.c
+++ b/kernel/softirq.c
@@ -846,7 +846,7 @@ __tasklet_common_schedule(struct tasklet_struct *t, struct tasklet_head *head, u
 		 */
 		else if (cmpxchg(&t->state, exp | TASKLET_STATEF_RUN,
 				 exp|TASKLET_STATEF_RUN|TASKLET_STATEF_SCHED)
-			 == exp | TASKLET_STATEF_RUN)
+			 == (exp | TASKLET_STATEF_RUN))
 			return;
 
 		/* If SCHED is already set, do nothing...
-- 
1.7.5.4

